### DISCUSSION TITLE
Integration of Web Frameworks and Costs of Usage in Startup Environment

### DISCUSSION THREAD

**Discussion by neatkent** [modular/modular#436] [Q&A] [answered]
Hello!

I'm trying to get started in coding professionally and I am investing time learning Django as a web framework. I am interested in having an idea of when and if Django will be fully integrated into Mojo language. Since the core functionality is implemented in python and it is some ~300,000 lines of code I can see why it might me pushed back,  but more importantly will it be worth it to the community at all to integrate such a large framework. Do you think that mojo will be so well adopted that learning a framework like Django or Flask might become an outdated skill if mojo doesn't adopt it or creates its own in a way in which the others are left only on legacy projects that are too costly to move over until they are not? 

In the entrepreneurial space.  Will mojo only be useful in building more efficient web apps that desire to integrate AI frameworks or will Mojo be useful to anyone wanting to make their code more efficient?

How will mojo influence developers financially in the future. Will there be a fee for the text editor? How will Modular price the usage of the software? Will there be a free version for companies that make under a financial threshold like Docker? How will usage be priced?  Is Modular already financial sustainable? How does Modular plan to be financially sustainable as an organization long term?

Thank you for your time!

**Comment by lattner** [comment, marked as answer]
Lots of questions here :-).  Yes, Mojo is generally useful for building out general purpose applications including django like things.  That said, whether such a thing happens depends a lot on the Mojo community.  I expect it to be vibrant and would love to help catalyze that, but it will depend on interest and motivation from a big group of folks.  Please stay tuned for updates on downloadable Mojo, it is coming soon, and we're aiming for somewhat higher than a compiler and vi support. :)

**Reply by neatkent** [reply]
Thank you Chriss Lattner!

What an honor!  I've been listening to your conversations with Lex and others on YouTube. I am really excited about Mojo and the future looks bright for Python! After exploring Modular's site a little, reading this: https://opensource.stackexchange.com/questions/13958/what-is-the-license-of-mojo, and projecting my thoughts from your response. I am wondering if Modular intends to keep open source mojo to the community, hopefully in return for community investment in "sharpening the tool" so to speak? I am grateful for the value you are adding to Python and the programming community as a whole. I am inspired, and I hope to have great positive impact myself someday!

Thank you!

**Reply by lattner** [reply]
We're figuring it out as we go.  We have a lot of basic stuff to implement right now, and are focused on enabling download of mojo. Stay tuned.

**Comment by Cafezinho** [comment]
Hello, in this preview phase SDK, what could be the features not available yet that could impact in the use of Mojo during web development like Flask or Django? I am eager to try it out, but I am expecting to get plenty of errors in this beggining. Am I wrong?

**Comment by ksandvik** [comment]
Various Python syntax constructs not yet in mojo like class, if the frameworks use C libraries (file and network io, libc and so on), no higher abstractions on top of the libc calls, no official way to call those from Mojo yet (all kinds of low-end workarounds). No way to call Mojo from Python yet. But it's worth trying out and file specific bug reports -- just make sure they are not duplicates. Also, I would expect this kind of work be isolated to parts that really need optimization than across the whe whole framework, initially.

There's also the option of starting a brand new web framework in Mojo only.
